core/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -751,6 +778,12 @@ Number valueForSearch(Number value) {
         }
 
         /**
+         * @throws IllegalArgumentException if value is not finite for this type
+         */
+        void validateParsed(Number value) {
+        }
```

Comment:
Since every numeric type should implement this method, can we have this throw an `UnsupportedOperationException` here so we get an error if we implement a new numeric type and forget to override this method?